### ISSUE TITLE
Error with code notation.

### ISSUE THREAD

**Issue by akanimax** [mkocabas/CoordConv-pytorch#12]
Hi @mkocabas,

There is a slight mismatch in the notation used to denote the extracted dimensions from the shape of the input tensor. [here](https://github.com/mkocabas/CoordConv-pytorch/blob/master/CoordConv.py#L84)

    batch_size, _, x_dim, y_dim = input_tensor.size()

You extract `x_dim` from dimension number `2` and `y_dim` from dimension number `3`. As per the image based `4d` tensors of Pytorch, it is the height that comes before width. Please refer to the documentation of [Pytorch's Conv2d layer](https://pytorch.org/docs/stable/nn.html?highlight=conv2d#torch.nn.Conv2d). Notice the input and output tensor shapes. 

The code would run fine since the extracted dimensions end up in right place when constructing the output tensor, but for a code reader like me (implementation checking), it seems like a mistake at first glance. Perhaps, the notation can be changed as per Pytorch.
Thanks

Cheers!
@akanimax

**Comment by CharlesNord** [comment]
> Hi @mkocabas,
> 
> There is a slight mismatch in the notation used to denote the extracted dimensions from the shape of the input tensor. [here](https://github.com/mkocabas/CoordConv-pytorch/blob/master/CoordConv.py#L84)
> 
> ```
> batch_size, _, x_dim, y_dim = input_tensor.size()
> ```
> You extract `x_dim` from dimension number `2` and `y_dim` from dimension number `3`. As per the image based `4d` tensors of Pytorch, it is the height that comes before width. Please refer to the documentation of [Pytorch's Conv2d layer](https://pytorch.org/docs/stable/nn.html?highlight=conv2d#torch.nn.Conv2d). Notice the input and output tensor shapes.
> 
> The code would run fine since the extracted dimensions end up in right place when constructing the output tensor, but for a code reader like me (implementation checking), it seems like a mistake at first glance. Perhaps, the notation can be changed as per Pytorch.
> Thanks
> 
> Cheers!
> @akanimax

I agree with you , the code works fine is because on line 95 and 96, the author applies .transpose(2, 3) to the xx_channel and yy_channel:
```
xx_channel = xx_channel.repeat(batch_size, 1, 1, 1).transpose(2, 3)
yy_channel = yy_channel.repeat(batch_size, 1, 1, 1).transpose(2, 3)
```

**Comment by akanimax** [comment]
@CharlesNord,

Yes the code works fine but I just pointed out that it is slightly confusing to read. Has it been updated yet? Let me know if you need any more help.

Best regards,
@akanimax
